src/main/java/org/jpeek/calculus/java/fix/Ccm.java

```diff
@@ -0,0 +1,286 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.calculus.java.fix;
+
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.jpeek.graph.java.FindConnectedComponents;
+
+/**
+ * Application.
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 1.0.0
+ * @checkstyle NestedIfDepthCheck (300 lines)
+ * @checkstyle MagicNumberCheck (300 lines)
+ * @checkstyle ExecutableStatementCountCheck (300 lines)
+ */
+public final class Ccm {
+
+    /**
+     * XML Skeleton.
+     */
+    private final XML skeleton;
+
+    /**
+     * The transformed XML skeleton.
+     */
+    private final XML tempres;
+
+    /**
+     * Initializes object of class for fixing NCC value of CCM metric.
+     *
+     * @param skeleton XML Skeleton
+     * @param tempres The Transformed XML skeleton.
+     */
+    public Ccm(final XML skeleton, final XML tempres) {
+        this.skeleton = skeleton;
+        this.tempres = tempres;
+    }
+
+    /**
+     * Updates the transformed xml with proper NCC value.
+     *
+     * @return XML with fixed NCC.
+     */
+    @SuppressWarnings("PMD.AvoidDuplicateLiterals")
+    public XML getFixedResult() {
+        final List<String> packages = this.skeleton.xpath("//package/@id");
+        String finalres = "";
+        if (this.tempres.toString().indexOf("<package") > 6) {
+            finalres = this.tempres.toString().substring(
+                0,
+                this.tempres.toString().indexOf("<package") - 6
+            );
+        }
+        for (final String apackage : packages) {
+            finalres = finalres.concat(
+                String.format(
+                    "      <package id=\"%s\">\n",
+                    apackage
+                )
+            );
+            final List<String> classes = this.skeleton.xpath(
+                String.format(
+                    "//package[@id ='%s']/class/@id",
+                    apackage
+                )
+            );
+            for (final String aclass : classes) {
+                finalres = this.calc(aclass, finalres);
+            }
+            finalres = finalres.concat("      </package>\n");
+        }
+        if (this.tempres.toString().indexOf("</app") > 1) {
```

Comment:
@iMaks99 this method of XML manipulations look wrong to me, since it's NOT canonical. What if there will `</app >` (which is also correct). Your code will break.

Reply:
@yegor256 
1) In case of `</app >` or anything differ from `</app` it will be okay. But our code works fine in these cases. 
2) new XMLDocument(`...`), where `...` is String value with `</app >` initialized before calling CCM metric calculation in XslCalculus returns `</app `.